src/main/java/com/google/cloud/spanner/hibernate/SpannerDialect.java

```diff
@@ -66,4 +69,79 @@ public boolean supportsUnionAll() {
   public boolean supportsCaseInsensitiveLike() {
     return false;
   }
+
+  /* DDL-related functions */
+
+  @Override
+  public boolean canCreateSchema() {
+    return false;
+  }
+
+  @Override
+  public String[] getCreateSchemaCommand(String schemaName) {
+    throw new UnsupportedOperationException(
+        "No create schema syntax supported by " + getClass().getName());
+  }
+
+  @Override
+  public String[] getDropSchemaCommand(String schemaName) {
+    throw new UnsupportedOperationException(
+        "No drop schema syntax supported by " + getClass().getName());
+  }
+
+  @Override
+  public String getCurrentSchemaCommand() {
+    throw new UnsupportedOperationException("No drop schema syntax supported by "
+        + getClass().getName());
+  }
+
+  @Override
+  public SchemaNameResolver getSchemaNameResolver() {
+    throw new UnsupportedOperationException(
+        "No schema name resolver supported by " + getClass().getName());
+  }
+
+  @Override
+  public boolean dropConstraints() {
+    return false;
```

Comment:
Would "DROP INDEX" apply, since index provides a uniqueness constraint?

Reply:
I don't think so here actually. The reason is Hibernate considers indexes and constraints two separate things and has a baked-in assumption that the UNIQUE word is associated with the "CONSTRAINT" word instead of the "INDEX" word (and has separate functions for indexes and constraints for us to override). What I'm getting at is that I think we need to say we don't support constraints and instead we need to drop the index when dropping table transparently to Hibernate. I have a feeling if we start crossing index and constraint wires with Hibernate it'll bite us later on.

Reply:
Right, I think that in a typical RDMBS the indexes are dropped automatically when you do `DROP TABLE`. Does Spanner not do that? Does it require you to drop indexes separately?

Reply:
Yes, in Spanner you must do this with separate statements.

Reply:
Yea that's right. https://cloud.google.com/spanner/docs/data-definition-language#drop_table look at the note in blue.

Reply:
interesting